docs/disk_cleaning.md

```diff
@@ -0,0 +1,140 @@
+CAPM3 machine controller copies `automatedCleaningMode` field value from a machine to its
+corresponding host. Note that machine takes precedence over host, so if
+the value of `automatedCleaningMode` doesn't match between host and machine, CAPM3 machine
+controller will edit the host. However, this is not always the case, see below for more info.
+
+#### When should I edit my CRs to disable cleaning
+
+Automated cleaning takes place when its first provisioning or during each deprovisioning of a
+host. You can disable cleaning before provisioning or deprovisioning. Disabling before
+deprovisioning means any data on the host will not be wiped out and kept as it is. Note that
+this may introduce security vulnerabilities if you have sensitive data which must be wiped out
+when the host is being recycled.
+
+#### What happens if I unset automatedCleaningMode in template
+
+`automatedCleaningMode` accepts `metadata` and `disabled` or it can be unset. Please note that,
+there is no default value in CAPM3 template and machine objects for `automatedCleaningMode`. So,
+if you don't set either metadata or disabled in the template, it will be unset both in the
+template and machine objects.
+
+**Note:** There is a default value, `metadata`, in the host CR. For that reason, even if it is
+unset in the template and machine objects, Ironic node will still face automated cleaning because
+the host has enabled it by its default value.
+
+#### Selective configuration
+
+If you have more than one machine referencing to the same template, then the `automatedCleaningMode`
+value set on the template will be copied to all those machines and eventually to corresponding hosts.
+As such, by default you can not have different configurations for two nodes if they are under the same
+pool. They both will have the same value, either `metadata` or `disabled`.
+
+However, it is possible to have cleaning disabled for one of the machine, while enabled for the other,
+even though they are in the same pool (i.e. referencing the same template). To achieve that:
+
+1. Don't set `automatedCleaningMode` in the template. If unset in the template, it will be unset in
+    machines too.
+
+1. Since unset in the machines, hosts will get its default value `metadata`.
+
+1. Now we have two hosts, both cleaning enabled. Since `automatedCleaningMode` is unset in the template,
+    you can set different cleaning modes for machines in the same pool. In other words, replicating the
+    value of `automatedCleaningMode` from template to machines is blocked because it is unset in the template.
+    This gives freedom to users to configure cleaning differently and selectively for each node regardless
+    of the fact that they belong to the same template.
+
+1. You can edit one of the machine to set `automatedCleaningMode` to `disabled`. Once edited,
+    CAPM3 machine controller replicates the value from the machine to its owning host. As such, you end up
```

Comment:
I would rather use metal3machine controller instead of CAPM3

Reply:
sure, I can fix it